osquery/tables/system/darwin/keychain_utils.cpp

```diff
@@ -133,6 +133,12 @@ std::string genSHA1ForCertificate(X509* cert) {
   return "";
 }
 
+std::string genSerialForCertificate(X509* cert) {
+  ASN1_INTEGER* serial = X509_get_serialNumber(cert);
+  BIGNUM* bignumSerial = ASN1_INTEGER_to_BN(serial, NULL);
+  return BN_bn2hex(bignumSerial);
```

Comment:
I think there’s a memory leak here. From the OpenSSL docs it looks like the caller is responsible for the return from both ASN1_Integrr_to_BN and BN_bn2hex, but I just skimmed the docs so I’m not positive.